x-pack/plugins/observability_solution/apm/server/routes/services/get_service_agent.ts

```diff
@@ -101,12 +102,17 @@ export async function getServiceAgent({
     return {};
   }
 
-  const { agent, service, cloud } = response.hits.hits[0]._source as ServiceAgent;
+  const { agent, service, cloud } = serviceAgentNameMapping(
+    response.hits.hits[0].fields
+  ) as ServiceAgent;
   const serverlessType = getServerlessTypeFromCloudData(cloud?.provider, cloud?.service?.name);
 
   return {
     agentName: agent?.name,
-    runtimeName: service?.runtime?.name,
+    runtimeName:
+      typeof service?.runtime?.name === 'string'
+        ? service?.runtime?.name
+        : service?.runtime?.name?.text,
```

Comment:
let's not do this, should be easy to fix in the mapper

Reply:
I did that and now that we have the mapper we don't need to do that anymore, thanks for the comment, it was already in my todo list :)